### PR TITLE
CFE-4073: Added file-permissions module

### PR DESCRIPTION
**Pull request by nickanderson**
First to publish a module with inputs!

### CODE REVIEW

File: cfbs.json [hunk 1]
```diff
@@ -139,6 +139,52 @@
...
+      "tags": ["management"],
+      "repo": "https://github.com/nickanderson/cfengine-file-permissions",
+      "by": "https://github.com/nickanderson",
+      "version": "0.0.1",
+      "commit": "a678160011b834557f1ab42a081ea84adf8622df",
+      "steps": [
+        "copy file-permissions.cf services/file-permissions.cf",
+        "input file-permissions/input.json def.json",
+        "bundles nickanderson:file_permissions",
+        "json inputs-def.json def.json"
```
Comment (olehermanse) [inline cfbs.json:153]: With the latest cfbs, you should be able to do:

```suggestion
        "policy_files services/file-permissions.cf"
```

(And delete your inputs-def.json from your repo)

File: cfbs.json [hunk 2]
```diff
@@ -139,6 +139,52 @@
...
+      ],
+      "input": [
+        {
+          "type": "list",
+          "variable": "files",
+          "namespace": "nickanderson",
+          "bundle": "file_permissions",
+          "label": "Filepaths",
```
Comment (olehermanse) [inline cfbs.json:161]:
```suggestion
          "label": "Files",
```

File: cfbs.json [hunk 3]
```diff
@@ -139,6 +139,52 @@
...
+          "subtype": [
+            {
+              "key": "path",
+              "type": "string",
+              "label": "Path",
+              "question": "Path of file?"
```
Comment (olehermanse) [inline cfbs.json:167]: If you don't use question mark, a colon will be added as prompt instead

```suggestion
              "question": "Path to file"
```

File: cfbs.json [hunk 4]
```diff
@@ -139,6 +139,52 @@
...
+            },
+            {
+              "key": "mode",
+              "type": "string",
+              "label": "Octal mode",
```
Comment (olehermanse) [inline cfbs.json:172]:
```suggestion
              "label": "Permission bits (octal)",
```

File: cfbs.json [hunk 5]
```diff
@@ -139,6 +139,52 @@
...
+              "question": "File permissions?",
```
Comment (olehermanse) [inline cfbs.json:173]:
```suggestion
              "question": "Permission bits (octal)",
```

File: cfbs.json [hunk 6]
```diff
@@ -139,6 +139,52 @@
...
+              "default": "600"
+            },
+            {
+              "key": "why",
+              "type": "string",
+              "label": "Why",
+              "question": "Why must these permissions be set so?",
```
Comment (olehermanse) [inline cfbs.json:180]:
```suggestion
              "question": "Why do these permissions matter?",
```

File: cfbs.json [hunk 7]
```diff
@@ -139,6 +139,52 @@
       "subdirectory": "management/every-minute",
       "steps": ["json def.json def.json"]
     },
+    "file-permissions": {
+      "description": "Manages file permissions",
+      "tags": ["management"],
+      "repo": "https://github.com/nickanderson/cfengine-file-permissions",
+      "by": "https://github.com/nickanderson",
+      "version": "0.0.1",
+      "commit": "a678160011b834557f1ab42a081ea84adf8622df",
```
Comment (olehermanse) [inline cfbs.json:148]: Update this if you end up making changes to your repo

File: cfbs.json [hunk 8]
```diff
@@ -139,6 +139,52 @@
...
+      "steps": [
+        "copy file-permissions.cf services/file-permissions.cf",
```
Comment (olehermanse) [inline cfbs.json:150]: I'd suggest moving the file to:
```suggestion
        "copy file-permissions.cf services/cfbs/modules/file-permissions/file-permissions.cf",
```

File: cfbs.json (hunk 1 above)
Comment (olehermanse) [reply cfbs.json:153]: Add in `services/cfbs/modules/file-permissions/file-permissions.cf` if you end up changing the path above.

File: cfbs.json [hunk 9]
```diff
@@ -139,6 +139,52 @@
...
+        "input file-permissions/input.json def.json",
+        "bundles nickanderson:file_permissions",
```
Comment (olehermanse) [inline cfbs.json:152]: I don't have a definitive suggestion on what the namespace should be, but perhaps just the canonified module name is okay?
```suggestion
        "bundles file_permissions:file_permissions",
```

**Review by olehermanse — COMMENTED** [review]
I tested it and it works well. Also made me discover and fix a bug in cfbs build steps. Added a bunch of comments, mostly nitpicks / discussion about what our recommendations should be for module authors.

File: cfbs.json (hunk 1 above)
Comment (nickanderson) [reply cfbs.json:153]: policy_files with no source specified?

File: cfbs.json (hunk 9 above)
Comment (nickanderson) [reply cfbs.json:152]: moving to file_permissions namespce makes me want to rename the bundle to unix or something.

**Pull request merged**